File: 1072. Flip Columns For Maximum Number of Equal Rows/Solution.py
from collections import Counter
from typing import List


class Solution:
    def maxEqualRowsAfterFlips(self, matrix: List[List[int]]) -> int:
        n = len(matrix[0])
        cnt = Counter()
        for row in matrix:
            if row[0]:
                for j in range(n):
                    row[j] ^= 1
            cnt[tuple(row)] += 1
        return max(cnt.values())

File: 1072. Flip Columns For Maximum Number of Equal Rows/Solution.cpp
#include <algorithm>
#include <string>
#include <unordered_map>
#include <vector>
using namespace std;

class Solution {
public:
    int maxEqualRowsAfterFlips(vector<vector<int>>& matrix)
    {
        int n = matrix[0].size(), res = 0;
        unordered_map<string, int> cnt;
        for (const vector<int>& row : matrix) {
            string s(n, 0);
            for (int j = 0; j < n; j++)
                s[j] = row[j] ^ row[0];
            res = max(res, ++cnt[s]);
        }
        return res;
    }
};
